x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskReplicationTests.java

```diff
@@ -57,240 +73,250 @@
 import java.util.function.LongConsumer;
 import java.util.stream.Collectors;
 
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 
 public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTestCase {
 
     public void testSimpleCcrReplication() throws Exception {
-        try (ReplicationGroup leaderGroup = createGroup(randomInt(2));
-             ReplicationGroup followerGroup = createFollowGroup(randomInt(2))) {
+        try (ReplicationGroup leaderGroup = createLeaderGroup(randomInt(2))) {
             leaderGroup.startAll();
-            int docCount = leaderGroup.appendDocs(randomInt(64));
-            leaderGroup.assertAllEqual(docCount);
-            followerGroup.startAll();
-            ShardFollowNodeTask shardFollowTask = createShardFollowTask(leaderGroup, followerGroup);
-            final SeqNoStats leaderSeqNoStats = leaderGroup.getPrimary().seqNoStats();
-            final SeqNoStats followerSeqNoStats = followerGroup.getPrimary().seqNoStats();
-            shardFollowTask.start(
+            try (ReplicationGroup followerGroup = createFollowGroup(leaderGroup, randomInt(2))) {
```

Comment:
I can open a separate PR to add a remote-recovery simulation to ShardFollowTaskReplicationTests.

Reply:
I opened #39007. You only have to look at `testRemoteRecoveryWithGapsInIndexCommit` in this file.